config/src/main/java/com/typesafe/config/impl/SerializedConfigValue.java

```diff
@@ -287,6 +290,16 @@ static SimpleConfigOrigin readOrigin(DataInput in, SimpleConfigOrigin baseOrigin
                 m.put(field, v);
         }
     }
+    
+   private static Collection<String> split(String value, int charSize) {
+        List<String> strings = new ArrayList<String>();
```

Comment:
The split function used only once and makes the excessive memory allocation here, looks like a copypasted from another library. Should it get the target collection as imput? Alternatively the calling code could be rewritten to remove ArrayList allocation.